code/modules/projectiles/gun_system.dm

```diff
@@ -665,9 +689,13 @@ and you're good to go.
 //----------------------------------------------------------
 
 /obj/item/weapon/gun/proc/Fire()
-	if(QDELETED(gun_user) || !ismob(gun_user) || !able_to_fire(gun_user) || !target)
+	if(QDELETED(gun_user) || !ismob(gun_user) || !target)
 		return
 
+	if(!is_deployed())
+		if(!able_to_fire(gun_user))
```

Comment:
this should be inline in the first check

if(QDELETED(gun_user) || !ismob(gun_user) || !target || (!is_deployed() && !able_to_fire(gun_user))
    return